lesson4/public/tickets.html

```diff
@@ -0,0 +1,90 @@
+<!DOCTYPE html>
+<html lang="en">
+<head>
+  <meta charset="UTF-8">
+  <title></title>
+</head>
+<body>
+  <p>Kinue's Tickets</p>
+  <pre id='display'></pre>
+  Tickets to buy:<input type='text' id='#Tickets'/>
+  <button type='button' id='theButton'>Buy me</button>
+  <pre id= 'display2'></pre>
+
+  <p>Saumya's Tickets</p>
+  <pre id= 'displaysya2'></pre>
+  How many tickets you need:<input type='text' id='syatick'/>
+  <input type='submit' id='syaButton'/>
+  <div>
+    <pre id= 'displaysya'></pre>
+  </div>
+  <script>
+    //send the # of tickets bought
+    const button = document.getElementById('theButton');
+    button.onclick = () =>{
+    const quantity = document.getElementById('#Tickets').value;
+    if(quantity<=0){
+    alert('Not valid entry');
+    }else{
+    const xhttp = new XMLHttpRequest();
+    xhttp.open('GET', `/santos?number=${quantity}`);
+    xhttp.onreadystatechange = () =>{
+      if(xhttp.responseText == 'Not enough tickets'){
+        alert('Not enough tickets');
+      }else{
+        const show1 = document.getElementById('display2');
+        show1.innerHTML = xhttp.responseText;
+      }
+    }
+    xhttp.send();
+    }
+    }
+    
+    //get update every second
+    const unseg= ()=>{
+      const xhttp1 = new XMLHttpRequest();
+      xhttp1.open('GET', `/update`);
+      xhttp1.send();
+      xhttp1.onreadystatechange = () =>{
+        if(xhttp1.responseText){         
+          const show = document.getElementById('display');
+          show.innerHTML = xhttp1.responseText;
+        }
+      }
+    }
+    setInterval(unseg, 1000);
+//SYA code
+//Button functionality to send data.
+
+const button2 = document.getElementById('syaButton');
+button2.onclick = () =>{
+  const tickets2 = document.getElementById('syatick').value;
+  const xhttp2 = new XMLHttpRequest();
+  xhttp2.open('GET', `http://sz.llip.life/buy?amnt=${tickets2}`);
+  xhttp2.onreadystatechange = () =>{
+    if(xhttp2.responseText){
+    const show2 = document.getElementById('displaysya');
+    show2.innerHTML = xhttp2.responseText;
+    }
+  }
+xhttp2.send()
+}
+
+//Function to update the number of Tickets
+
+const update2 = () =>{
+  const xhttp3 = new XMLHttpRequest();
```

Comment:
you can put all these into unseg function. This way, you don't need to have 2 setIntervals in your code.